Automattic/jetpack
# Masterbar: Disable Calypsoify nav overrides when Nav Unification is enabled

#### Changes proposed in this Pull Request:
Calypsoify is used for (1) overriding the WP Admin navigation to make it more similar to the Calypso navigation and (2) for injecting scripts and styles needed by the block editor when accessed from Calypso.

This diff ensures the custom overrides done by Calypsoify to the navigation do not take place when the Nav Unification is enabled, so the custom overrides done by the Nav Unification are preserved.

The Calypsoify customizations for the block editor are not modified by these changes.

**Note**: This only affects Jetpack and Atomic sites sites. Simples sites use a different version of Calypsoify and they are covered on D55269-code.

#### Jetpack product discussion
pbAPfg-VL-p2#comment-1904

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Make sure your site is connected to WP.com.
- Make sure the Nav Unification is disabled (it is disabled by default, unless you have registered a `jetpack_load_admin_menu_class` filter for enabling it).
- Go to WP Admin > Jetpack > Settings > Writing, and enable the WordPress.com toolbar: <img width="1062" alt="Screen Shot 2021-01-13 at 16 03 12" src="https://user-images.githubusercontent.com/1233880/104469487-dc1a8100-55b8-11eb-9164-7c0069f2aa1c.png">
- Visit a Calypsoified WP Admin page. Example: `/wp-admin/edit.php?post_type=feedback&calypsoify=1`.
- Make sure the navigation has been overridden by Calypsoify: <img width="1271" alt="Screen Shot 2021-01-13 at 16 04 05" src="https://user-images.githubusercontent.com/1233880/104469603-fb191300-55b8-11eb-9a8e-4449f08bf017.png">
- Enable the Nav Unification by adding a mu-plugin that contains: `add_filter( 'jetpack_load_admin_menu_class', '__return_true' );`.
- Visit the same page that you Calypsoified above.
- Make sure the navigation has been overridden by the Nav Unification, and the Calypsoify overrides didn't take place: <img width="1272" alt="Screen Shot 2021-01-13 at 16 05 50" src="https://user-images.githubusercontent.com/1233880/104469822-39aecd80-55b9-11eb-82fc-f69ba973b63b.png">
- Load the Calypsoified block editor (`/wp-admin/post-new.php?calypsoify=1`).
- Open the browser JS console and make sure the `calypsoifyGutenberg` var is defined ([this var is set by Calypsoify](https://github.com/Automattic/jetpack/blob/122c45b155dc40864061f0f6449e0e60afcea360/modules/calypsoify/class-jetpack-calypsoify.php#L215-L223)): <img width="1250" alt="Screen Shot 2021-01-13 at 16 07 40" src="https://user-images.githubusercontent.com/1233880/104470009-7b3f7880-55b9-11eb-88ac-e68869e3797a.png">

#### Proposed changelog entry for your changes:
N/A.
